app-editors/vscodium/vscodium-1.56.2.ebuild

```diff
@@ -0,0 +1,94 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop pax-utils xdg
+
+DESCRIPTION="A community-driven, freely-licensed binary distribution of Microsoft's VSCode"
+HOMEPAGE="https://vscodium.com/"
+SRC_URI="
+	amd64? ( https://github.com/VSCodium/${PN}/releases/download/${PV}/VSCodium-linux-x64-${PV}.tar.gz -> ${P}-amd64.tar.gz )
+	arm? ( https://github.com/VSCodium/${PN}/releases/download/${PV}/VSCodium-linux-armhf-${PV}.tar.gz -> ${P}-arm.tar.gz )
+	arm64? ( https://github.com/VSCodium/${PN}/releases/download/${PV}/VSCodium-linux-arm64-${PV}.tar.gz -> ${P}-arm64.tar.gz )
+"
+
+RESTRICT="mirror strip bindist"
+
+LICENSE="
+	Apache-2.0
+	BSD
+	BSD-1
+	BSD-2
+	BSD-4
+	CC-BY-4.0
+	ISC
+	LGPL-2.1+
+	MIT
+	MPL-2.0
+	openssl
+	PYTHON
+	TextMate-bundle
+	Unlicense
+	UoI-NCSA
+	W3C
+"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~arm ~arm64"
+IUSE=""
+
+RDEPEND="
+	app-accessibility/at-spi2-atk
+	app-crypt/libsecret[crypt]
+	dev-libs/nss
+	media-libs/alsa-lib
+	media-libs/libpng:0/16
+	x11-libs/cairo
+	x11-libs/gtk+:3
+	x11-libs/libnotify
+	x11-libs/libxkbcommon
+	x11-libs/libxkbfile
+	x11-libs/libXScrnSaver
+	x11-libs/libXtst
+	x11-libs/pango
+"
```

Comment:
```
 * QA Notice: Unresolved soname dependencies:
 * 
 * 	/opt/vscodium/codium: libcups.so.2
 * 
```
```
$ vscodium 
/opt/vscodium/bin/../codium: error while loading shared libraries: libcups.so.2: cannot open shared object file: No such file or directory
```
Works fine after installing cups.

Reply:
Does that mean `vscode` needs it, is there a simple way to get these missing soname dependencies ? Or do I need a clean chroot ?

Reply:
I'm not sure why I didn't hit this error with *vscode*. Could you please provide the list for: 
*/var/db/pkg/app-editors/vscode-version/NEEDED* and */var/db/pkg/app-editors/vscode-version/NEEDED.ELF.2*

Yes, a clean stage3 chroot is usually the most accurate way of testing your deps. You'll need `FEATURES="qa-unresolved-soname-deps` to catch them.

Then you can compare the above file I gave you with your RDEPEND list after it's installed, strace / scanefl binaries and libraries...
```#!/bin/bash

qlist -e ${1} \
	| xargs --no-run-if-empty scanelf -L -n -q -F '%n #F' \
	| tr , '\n' \
	| xargs --no-run-if-empty readlink -f \
	| uniq \
	| xargs --no-run-if-empty qfile -CS \
	| sort -u
```

Reply:
> Could you please provide the list for:
/var/db/pkg/app-editors/vscode-version/NEEDED and /var/db/pkg/app-editors/vscode-version/NEEDED.ELF.2

You mean the ones I have on my machine right now ? I attached them.

[vscode-NEEDED.zip](https://github.com/gentoo/gentoo/files/6566293/vscode-NEEDED.zip)

> Yes, a clean stage3 chroot is usually the most accurate way of testing your deps. You'll need FEATURES="qa-unresolved-soname-deps to catch them.

Alright, I will set that up for the next times. I liked the BTRFS snapshot approach when I was reading of the various ways to do it.